src/components/access-type.spec.js

```diff
@@ -0,0 +1,254 @@
+            return <div />;
+          }}
+        </AccessType>
+      </Provider>
+    );
+    expect(await instance(selectedPlan, "standard")).toStrictEqual({
+      type: "standard",
+      plan: {
+        id: 47742,
+        title: "Selected Plan",
+        description: "Selected Plan",
+        price_cents: 3000,
+        price_currency: "THB",
+        duration_length: 5,
+        duration_unit: "days"
+      },
+      coupon_code: "",
+      payment: {
+        payment_type: "omise_recurring",
+        amount_cents: 3000,
+        amount_currency: "THB"
+      },
+      assets: [{ id: "", title: "", slug: "" }],
+      recipient_subscriber: {}
+    });
+  });
+  it("Successful omise one-time payment", async () => {
+    const container = render(
+      <Provider
+        store={createStore(store => store, {
+          paymentOptions: {
+            omise: {
+              action: "pay",
+              proceed: async paymentOptions => {
+                return { proceed: paymentObject => paymentObject };
+              }
+            }
+          }
+        })}
+      >
+        <AccessType
+          isStaging={true}
+          enableAccesstype={true}
+          accessTypeKey="dummyKey"
+          email="r@gmail.com"
+          phone="9900990099"
+          disableMetering={false}
+          stagingHost="https://staging.accesstype.com"
+          accessTypeBkIntegrationId="8"
+        >
+          {({ initOmisePayment }) => {
+            instance = initOmisePayment;
+            return <div />;
+          }}
+        </AccessType>
+      </Provider>
+    );
+    expect(await instance({ ...selectedPlan, recurring: false }, "standard")).toStrictEqual({
+      type: "standard",
+      plan: {
+        id: 47742,
+        title: "Selected Plan",
+        description: "Selected Plan",
+        price_cents: 3000,
+        price_currency: "THB",
+        duration_length: 5,
+        duration_unit: "days"
+      },
+      coupon_code: "",
+      payment: {
+        payment_type: "omise",
+        amount_cents: 3000,
+        amount_currency: "THB"
+      },
+      assets: [{ id: "", title: "", slug: "" }],
+      recipient_subscriber: {}
+    });
+  });
+  it("Failed omise payment when the payment options are not passed", async () => {
+    const container = render(
+      <Provider store={createStore(store => store, {})}>
+        <AccessType
+          isStaging={true}
+          enableAccesstype={true}
+          accessTypeKey="dummyKey"
+          email="r@gmail.com"
+          phone="9900990099"
+          disableMetering={false}
+          stagingHost="https://staging.accesstype.com"
+          accessTypeBkIntegrationId="8"
+        >
+          {({ initOmisePayment }) => {
+            instance = initOmisePayment;
+            return <div />;
+          }}
+        </AccessType>
+      </Provider>
+    );
+
+    await expect(instance(selectedPlan, "standard")).rejects.toEqual({
+      message: "Payment option is loading..."
+    });
```

Comment:
Can't await like this, we need to check the result of the promise instead of reaching into `.reject`